libvast/src/system/index.cpp

```diff
@@ -793,12 +793,40 @@ void index_state::send_report() {
        max_concurrent_partition_lookups - running_partition_lookups},
       {"scheduler.partition.current-lookups", running_partition_lookups},
     }};
+
+  msg.data.push_back(data_point{
+          .key = "memory-usage",
+          .value = memusage(),
+          .metadata = {
+            {"component", std::string{name}},
+          },
+        });
   self->send(accountant, std::move(msg));
   auto r = performance_report{.data = {{{"scheduler", scheduler_measurement}}}};
   self->send(accountant, std::move(r));
   scheduler_measurement = measurement{};
 }
 
+std::size_t index_state::memusage() const {
+  auto calculate_usage = []<class T>(const T& collection) -> std::size_t {
+    return collection.size() * sizeof(typename T::value_type);
+  };
+  auto usage = std::size_t{0u};
+  for (const auto& [type, partition_info] : active_partitions) {
+    usage += as_bytes(type).size() + sizeof(partition_info);
+  }
+  usage += persisted_partitions.size()
+           * sizeof(decltype(persisted_partitions)::key_type);
```

Comment:
```suggestion
           * sizeof(decltype(persisted_partitions)::value_type);
```
I didn't know `unordered_set` had a `key_type`, but even then, `value_type` seems still more appropriate.